File: Problemas/1873A - Short Sort/1873A_Short_Sort.cpp
// 1873A - Short Sort
// https://codeforces.com/problemset/problem/1873/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas = Número de casos de prueba a realizar
    int pruebas;
    cin >> pruebas;

    // Itera sobre cada caso de prueba
    for(int indice_prueba = 0; indice_prueba < pruebas; indice_prueba++) {
        
            // Input:
            // 1) cartas = Cartas (escritas en el orden que están)
            string cartas;
            cin >> cartas;

            // Variables auxiliares para realizar los intercambios de cartas
            string swap01 = cartas;
            string swap12 = cartas;
            string swap02 = cartas;

            // Prueba todas las combinaciones de intercambio de cartas
            // Guarda los resultados en las variables auxiliares
            swap(swap01[0], swap01[1]);
            swap(swap12[1], swap12[2]);
            swap(swap02[0], swap02[2]);
            
            // Checa si alguna de las combinaciones de intercambio da 'abc' como resultado
            // Checa si las cartas dan 'abc' tal como estaban ordenadas inicialmente
            // Si alguna combinación da 'abc', entonces imprime 'YES', de lo contrario, imprime 'NO'
            if(swap01 == "abc" || swap12 == "abc" || swap02 == "abc" || cartas == "abc") {
                cout << "YES" << endl;
            } else {
                cout << "NO" << endl;
            }
    }

    return 0;
}

File: Problemas/1873A - Short Sort/1873A_Short_Sort.py
# 1873A - Short Sort
# https://codeforces.com/problemset/problem/1879/A

# intercambiar_cartas: Intercambia dos cartas de cualquier posición en un string de cartas
# Input:
#   1) cartas: String de cartas a modificar
#   2) localidad1: Localidad de la 1° carta
#   3) localidad2: Localidad de la 2° carta
# Output:
#   cartas_intercambiadas: String con las dos cartas intercambiadas
def intercambiar_cartas(cartas: str, localidad1: int, localidad2: int):

    # Separa las cartas en una lista
    letras = list(cartas)

    # Intercambia las cartas seleccionadas
    letras[localidad1], letras[localidad2] = letras[localidad2], letras[localidad1]

    # Reúne las cartas en un solo string
    cartas_intercambiadas = "".join(letras)  

    # Devuelve las cartas intercambiadas
    return cartas_intercambiadas

# Input:
# 1) pruebas = Número de casos de prueba
pruebas = int(input())

# Itera sobre cada caso de prueba
for indice_prueba in range(0,pruebas):

    # Input:
    # 2) cartas = Cartas (escritas en el orden que están)
    cartas = input()

    # Checa si alguna de las combinaciones de intercambio da 'abc' como resultado
    # Checa si las cartas dan 'abc' tal como estaban ordenadas inicialmente
    # Si alguna combinación da 'abc', entonces imprime 'YES', de lo contrario, imprime 'NO'
    if intercambiar_cartas(cartas,0,1) == 'abc' or intercambiar_cartas(cartas,0,2) == 'abc' or intercambiar_cartas(cartas,1,2) == 'abc' or cartas == 'abc':
        print('YES')
    else:
        print('NO')
